File: app/src/main/java/net/kibotu/kalmanrx/app/misc/SensorDelay.java
package net.kibotu.kalmanrx.app.misc;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import static android.hardware.SensorManager.SENSOR_DELAY_FASTEST;
import static android.hardware.SensorManager.SENSOR_DELAY_GAME;
import static android.hardware.SensorManager.SENSOR_DELAY_NORMAL;
import static android.hardware.SensorManager.SENSOR_DELAY_UI;

@IntDef({SENSOR_DELAY_FASTEST, SENSOR_DELAY_GAME, SENSOR_DELAY_UI, SENSOR_DELAY_NORMAL})
@Retention(RetentionPolicy.SOURCE)
public @interface SensorDelay {
}

File: app/src/main/java/net/kibotu/kalmanrx/app/misc/SensorDelay.kt
package net.kibotu.kalmanrx.app.misc


import android.hardware.SensorManager.*
import androidx.annotation.IntDef

/**
 * Created by [Jan Rabe](https://about.me/janrabe).
 */

@IntDef(SENSOR_DELAY_FASTEST, SENSOR_DELAY_GAME, SENSOR_DELAY_UI, SENSOR_DELAY_NORMAL)
@Retention(AnnotationRetention.SOURCE)
annotation class SensorDelay